Crash during mvn io.quarkus-maven-plugin:2.1.2.Final:create

### Describe the bug

The maven plugin crashes in Windows Powershell when I try to create a new project. 



### Expected behavior

_No response_

### Actual behavior

The maven process crashes with the following exception:

```
com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) Error injecting constructor, java.lang.NoSuchMethodError: 'java.io.OutputStream org.fusesource.jansi.AnsiConsole.wrapOutputStream(java.io.OutputStream)'
  at io.quarkus.maven.components.Prompter.<init>(Unknown Source)
  while locating io.quarkus.maven.components.Prompter
  while locating io.quarkus.maven.CreateProjectMojo

...
```

### How to Reproduce?

Open up a Powershell instance.
Then execute the following command:

```
mvn io.quarkus:quarkus-maven-plugin:2.1.2.Final:create `
-DprojectGroupId="com.infosupport.pitstop.customermanagement" `
-DprojectArtifactId="customer-management" `
-DclassName="com.infosupport.pitstop.customermanagement.CustomersResource" `
-Dpath="customer-management"clear
```

### Output of `uname -a` or `ver`

Microsoft Windows [Version 10.0.19043.1110]

### Output of `java -version`

openjdk version "16.0.1" 2021-04-20 OpenJDK Runtime Environment AdoptOpenJDK-16.0.1+9 (build 16.0.1+9) OpenJDK 64-Bit Server VM AdoptOpenJDK-16.0.1+9 (build 16.0.1+9, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.1.2.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.2 (ea98e05a04480131370aa0c110b8c54cf726c06f)

### Additional information

Powershell version: 5.1.19041.1023

For the sake of reference for others, here's the actual stacktrace it generates:

```
Caused by: java.lang.NoSuchMethodError: 'java.io.OutputStream org.fusesource.jansi.AnsiConsole.wrapOutputStream(java.io.OutputStream)'
    at jline.AnsiWindowsTerminal.detectAnsiSupport (AnsiWindowsTerminal.java:57)
    at jline.AnsiWindowsTerminal.<init> (AnsiWindowsTerminal.java:27)
    at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0 (Native Method)
    at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance (NativeConstructorAccessorImpl.java:78)
    at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance (DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstanceWithCaller (Constructor.java:499)
    at java.lang.reflect.ReflectAccess.newInstance (ReflectAccess.java:128)
    at jdk.internal.reflect.ReflectionFactory.newInstance (ReflectionFactory.java:350)
    at java.lang.Class.newInstance (Class.java:642)
    at jline.TerminalFactory.getFlavor (TerminalFactory.java:205)
    at jline.TerminalFactory.create (TerminalFactory.java:96)
    at jline.TerminalFactory.get (TerminalFactory.java:180)
    at jline.TerminalFactory.get (TerminalFactory.java:186)
    at jline.console.ConsoleReader.<init> (ConsoleReader.java:244)
    at jline.console.ConsoleReader.<init> (ConsoleReader.java:236)
    at jline.console.ConsoleReader.<init> (ConsoleReader.java:224)
    at io.quarkus.devtools.utils.Prompter.<init> (Prompter.java:20)
    at io.quarkus.maven.components.Prompter.<init> (Prompter.java:17)
    at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0 (Native Method)
    at jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance (NativeConstructorAccessorImpl.java:78)
    at jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance (DelegatingConstructorAccessorImpl.java:45)
    at java.lang.reflect.Constructor.newInstanceWithCaller (Constructor.java:499)
    at java.lang.reflect.Constructor.newInstance (Constructor.java:480)
    at com.google.inject.internal.DefaultConstructionProxyFactory$ReflectiveProxy.newInstance (DefaultConstructionProxyFactory.java:126)
    at com.google.inject.internal.ConstructorInjector.provision (ConstructorInjector.java:114)
    at com.google.inject.internal.ConstructorInjector.access$000 (ConstructorInjector.java:32)
    at com.google.inject.internal.ConstructorInjector$1.call (ConstructorInjector.java:98)
    at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:112)
    at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:127)
    at com.google.inject.internal.ProvisionListenerStackCallback.provision (ProvisionListenerStackCallback.java:66)
    at com.google.inject.internal.ConstructorInjector.construct (ConstructorInjector.java:93)
    at com.google.inject.internal.ConstructorBindingImpl$Factory.get (ConstructorBindingImpl.java:306)
    at com.google.inject.internal.InjectorImpl$1.get (InjectorImpl.java:1050)
    at org.eclipse.sisu.inject.Guice4$1.get (Guice4.java:162)
    at org.eclipse.sisu.inject.LazyBeanEntry.getValue (LazyBeanEntry.java:81)
    at org.eclipse.sisu.plexus.LazyPlexusBean.getValue (LazyPlexusBean.java:51)
    at org.eclipse.sisu.plexus.PlexusRequirements$RequirementProvider.get (PlexusRequirements.java:250)
    at org.eclipse.sisu.plexus.ProvidedPropertyBinding.injectProperty (ProvidedPropertyBinding.java:48)
    at org.eclipse.sisu.bean.BeanInjector.injectMembers (BeanInjector.java:52)
    at com.google.inject.internal.MembersInjectorImpl.injectMembers (MembersInjectorImpl.java:160)
    at com.google.inject.internal.ConstructorInjector.provision (ConstructorInjector.java:124)
    at com.google.inject.internal.ConstructorInjector.access$000 (ConstructorInjector.java:32)
    at com.google.inject.internal.ConstructorInjector$1.call (ConstructorInjector.java:98)
    at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:112)
    at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:127)
    at com.google.inject.internal.ProvisionListenerStackCallback.provision (ProvisionListenerStackCallback.java:66)
    at com.google.inject.internal.ConstructorInjector.construct (ConstructorInjector.java:93)
    at com.google.inject.internal.ConstructorBindingImpl$Factory.get (ConstructorBindingImpl.java:306)
    at com.google.inject.internal.InjectorImpl$1.get (InjectorImpl.java:1050)
    at com.google.inject.internal.InjectorImpl.getInstance (InjectorImpl.java:1086)
    at org.eclipse.sisu.space.AbstractDeferredClass.get (AbstractDeferredClass.java:48)
    at com.google.inject.internal.ProviderInternalFactory.provision (ProviderInternalFactory.java:85)
    at com.google.inject.internal.InternalFactoryToInitializableAdapter.provision (InternalFactoryToInitializableAdapter.java:57)
    at com.google.inject.internal.ProviderInternalFactory$1.call (ProviderInternalFactory.java:66)
    at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:112)
    at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision (ProvisionListenerStackCallback.java:127)
    at com.google.inject.internal.ProvisionListenerStackCallback.provision (ProvisionListenerStackCallback.java:66)
    at com.google.inject.internal.ProviderInternalFactory.circularGet (ProviderInternalFactory.java:61)
    at com.google.inject.internal.InternalFactoryToInitializableAdapter.get (InternalFactoryToInitializableAdapter.java:47)
    at com.google.inject.internal.InjectorImpl$1.get (InjectorImpl.java:1050)
    at org.eclipse.sisu.inject.Guice4$1.get (Guice4.java:162)
    at org.eclipse.sisu.inject.LazyBeanEntry.getValue (LazyBeanEntry.java:81)
    at org.eclipse.sisu.plexus.LazyPlexusBean.getValue (LazyPlexusBean.java:51)
    at org.codehaus.plexus.DefaultPlexusContainer.lookup (DefaultPlexusContainer.java:263)
    at org.codehaus.plexus.DefaultPlexusContainer.lookup (DefaultPlexusContainer.java:255)
    at org.apache.maven.plugin.internal.DefaultMavenPluginManager.getConfiguredMojo (DefaultMavenPluginManager.java:520)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:124)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:972)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:78)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:567)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
```

--- a/build-parent/pom.xml
+++ b/build-parent/pom.xml
@@ -503,6 +503,10 @@
                                             <exclude>org.jboss.resteasy:resteasy-context-propagation</exclude>
                                             <exclude>com.google.android:annotations</exclude>
                                             <exclude>org.codehaus.mojo:animal-sniffer-annotations</exclude>
+                                            <!-- JLine 2.x (EOLed) version shouldn't be pulled in,
+                                            to prevent version compatibility issues with the
+                                            Jansi library shipped in Maven -->
+                                            <exclude>jline:jline</exclude>
                                         </excludes>
                                         <includes>
                                             <!-- this is for REST Assured -->

--- a/devtools/maven/src/main/java/io/quarkus/maven/components/Prompter.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/components/Prompter.java
@@ -1,8 +1,6 @@
 package io.quarkus.maven.components;
 
 import java.io.IOException;
-import java.io.InputStream;
-import java.io.OutputStream;
 
 import org.codehaus.plexus.component.annotations.Component;
 
@@ -17,7 +15,4 @@ public class Prompter extends io.quarkus.devtools.utils.Prompter {
         super();
     }
 
-    public Prompter(InputStream in, OutputStream out) throws IOException {
-        super(in, out);
-    }
 }

--- a/independent-projects/arc/pom.xml
+++ b/independent-projects/arc/pom.xml
@@ -245,6 +245,10 @@
                                             <exclude>org.slf4j:slf4j-jdk14</exclude>
                                             <exclude>org.slf4j:slf4j-log4j12</exclude>
                                             <exclude>org.slf4j:slf4j-log4j13</exclude>
+                                            <!-- JLine 2.x (EOLed) version shouldn't be pulled in,
+                                            to prevent version compatibility issues with the
+                                            Jansi library shipped in Maven -->
+                                            <exclude>jline:jline</exclude>
                                         </excludes>
                                     </bannedDependencies>
                                 </rules>

--- a/independent-projects/bootstrap/pom.xml
+++ b/independent-projects/bootstrap/pom.xml
@@ -123,6 +123,10 @@
                                             <exclude>org.slf4j:slf4j-jdk14</exclude>
                                             <exclude>org.slf4j:slf4j-log4j12</exclude>
                                             <exclude>org.slf4j:slf4j-log4j13</exclude>
+                                            <!-- JLine 2.x (EOLed) version shouldn't be pulled in,
+                                            to prevent version compatibility issues with the
+                                            Jansi library shipped in Maven -->
+                                            <exclude>jline:jline</exclude>
                                         </excludes>
                                     </bannedDependencies>
                                 </rules>

--- a/independent-projects/qute/pom.xml
+++ b/independent-projects/qute/pom.xml
@@ -173,6 +173,10 @@
                                             <exclude>org.slf4j:slf4j-jdk14</exclude>
                                             <exclude>org.slf4j:slf4j-log4j12</exclude>
                                             <exclude>org.slf4j:slf4j-log4j13</exclude>
+                                            <!-- JLine 2.x (EOLed) version shouldn't be pulled in,
+                                            to prevent version compatibility issues with the
+                                            Jansi library shipped in Maven -->
+                                            <exclude>jline:jline</exclude>
                                         </excludes>
                                     </bannedDependencies>
                                 </rules>

--- a/independent-projects/resteasy-reactive/pom.xml
+++ b/independent-projects/resteasy-reactive/pom.xml
@@ -364,6 +364,10 @@
                                             <exclude>org.slf4j:slf4j-jdk14</exclude>
                                             <exclude>org.slf4j:slf4j-log4j12</exclude>
                                             <exclude>org.slf4j:slf4j-log4j13</exclude>
+                                            <!-- JLine 2.x (EOLed) version shouldn't be pulled in,
+                                            to prevent version compatibility issues with the
+                                            Jansi library shipped in Maven -->
+                                            <exclude>jline:jline</exclude>
                                         </excludes>
                                     </bannedDependencies>
                                 </rules>

--- a/independent-projects/tools/devtools-common/pom.xml
+++ b/independent-projects/tools/devtools-common/pom.xml
@@ -44,7 +44,7 @@
         </dependency>
         <!-- user prompt -->
         <dependency>
-            <groupId>jline</groupId>
+            <groupId>org.jline</groupId>
             <artifactId>jline</artifactId>
         </dependency>
         <dependency>

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/utils/Prompter.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/utils/Prompter.java
@@ -1,11 +1,12 @@
 package io.quarkus.devtools.utils;
 
 import java.io.IOException;
-import java.io.InputStream;
-import java.io.OutputStream;
 import java.util.Objects;
-import jline.console.ConsoleReader;
 import org.apache.commons.lang3.StringUtils;
+import org.jline.reader.LineReader;
+import org.jline.reader.LineReaderBuilder;
+import org.jline.terminal.Terminal;
+import org.jline.terminal.TerminalBuilder;
 
 /**
  * Prompt implementation.
@@ -14,22 +15,15 @@ import org.apache.commons.lang3.StringUtils;
  */
 public class Prompter {
 
-    private final ConsoleReader console;
+    private final LineReader lineReader;
 
     public Prompter() throws IOException {
-        this.console = new ConsoleReader();
-        console.setHistoryEnabled(false);
-        console.setExpandEvents(false);
-    }
-
-    public Prompter(InputStream in, OutputStream out) throws IOException {
-        this.console = new ConsoleReader(in, out);
-        console.setHistoryEnabled(false);
-        console.setExpandEvents(false);
-    }
-
-    public ConsoleReader getConsole() {
-        return console;
+        // we set dumb to "true" only to prevent any warning when a proper terminal cannot be detected.
+        // If a proper terminal is detected by JLine then that terminal will be used and setting dumb=true
+        // won't force a dumb terminal.
+        // (https://github.com/jline/jline3/issues/291)
+        final Terminal terminal = TerminalBuilder.builder().dumb(true).build();
+        this.lineReader = LineReaderBuilder.builder().terminal(terminal).build();
     }
 
     public String prompt(final String message, final Character mask) throws IOException {
@@ -38,7 +32,7 @@ public class Prompter {
         final String prompt = String.format("%s: ", message);
         String value;
         do {
-            value = console.readLine(prompt, mask);
+            value = lineReader.readLine(prompt, mask);
         } while (StringUtils.isBlank(value));
         return value;
     }
@@ -53,7 +47,7 @@ public class Prompter {
         Objects.requireNonNull(defaultValue);
 
         final String prompt = String.format("%s [%s]: ", message, defaultValue);
-        String value = console.readLine(prompt);
+        String value = lineReader.readLine(prompt);
         if (StringUtils.isBlank(value)) {
             return defaultValue;
         }

--- a/independent-projects/tools/pom.xml
+++ b/independent-projects/tools/pom.xml
@@ -38,7 +38,7 @@
         <scala-plugin.version>4.4.0</scala-plugin.version>
 
         <!-- Versions -->
-        <jline.version>2.14.6</jline.version>
+        <jline.version>3.20.0</jline.version>
         <assertj.version>3.20.2</assertj.version>
         <jackson.version>2.12.4</jackson.version>
         <jakarta.enterprise.cdi-api.version>2.0.2</jakarta.enterprise.cdi-api.version>
@@ -130,7 +130,7 @@
                 <version>${project.version}</version>
             </dependency>
             <dependency>
-                <groupId>jline</groupId>
+                <groupId>org.jline</groupId>
                 <artifactId>jline</artifactId>
                 <version>${jline.version}</version>
             </dependency>
@@ -309,6 +309,10 @@
                                         <exclude>org.slf4j:slf4j-jdk14</exclude>
                                         <exclude>org.slf4j:slf4j-log4j12</exclude>
                                         <exclude>org.slf4j:slf4j-log4j13</exclude>
+                                        <!-- JLine 2.x (EOLed) version shouldn't be pulled in,
+                                        to prevent version compatibility issues with the
+                                        Jansi library shipped in Maven -->
+                                        <exclude>jline:jline</exclude>
                                     </excludes>
                                 </bannedDependencies>
                             </rules>

